collector/cluster_settings_test.go

```diff
@@ -35,10 +41,48 @@ func TestClusterSettingsStats(t *testing.T) {
 			if err != nil {
 				t.Fatalf("Failed to fetch or decode cluster settings stats: %s", err)
 			}
-			t.Logf("[%s/%s] Cluster Settings Stats Response: %+v", hn, ver, nsr)
+			t.Logf("[%s/%s] Cluster Settings Stats Response: %+v", hn, filename, nsr)
 			if nsr.Cluster.Routing.Allocation.Enabled != "ALL" {
 				t.Errorf("Wrong setting for cluster routing allocation enabled")
 			}
+			if nsr.Cluster.MaxShardsPerNode != "" {
+				t.Errorf("MaxShardsPerNode should be empty on older releases")
+			}
+		}
+	}
+}
+
+func TestClusterMaxShardsPerNode(t *testing.T) {
+	// Testcases created using:
+	//  docker run -d -p 9200:9200 elasticsearch:VERSION-alpine
+	//  curl http://localhost:9200/_cluster/settings/?include_defaults=true
+	files := []string{"../fixtures/settings-7.3.0.json"}
+	buf := bytes.NewBuffer(nil)
```

Comment:
same here

Reply:
Hm ok, still smells like a case of don't repeat yourself to me but if you can live with that... ;)